addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -40,6 +42,28 @@ options.UserValueWidget.include({
     },
 });
 
+Many2oneUserValueWidget.include({
+    /**
+     * @override
+     */
+    async _getSearchDomain() {
+        // We add the current website's domain if the model has a website_id
+        // field. Note that the rpcs are cached but we could get all models that
+        // have a website_id field in a single rpc(TODO).
+        const websiteIdField = await this._rpc({
+            model: this.options.model,
+            method: "fields_get",
+            args: [["website_id"]],
+        });
+        const modelHasWebsiteId = !!websiteIdField["website_id"];
+        if (modelHasWebsiteId && !this.options.domain.find(arr => arr[0] === "website_id")) {
+            this.options.domain =
+                Domain.and([this.options.domain, wUtils.websiteDomain(this)]).toList();
```

Comment:
Not sure I am fan to alter the `options.domain` instead of just returning it concat to the website domain but it conceptually also makes sense as it's the system adding it automatically.
I don't have strong opinion so :+1: